dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverImpl.java

```diff
@@ -813,14 +819,19 @@ void primeSelected(Collection<String> features) {
                     if (selectedChain != null) {
                         // TODO Need to revisit why this is not always a conflict.
                         // We only keep the first selected one
-                        iFeatures.remove();
+                        boolean allowedConflict = allowMultipleVersions(base);
+                        if (!allowedConflict) {
+                            iFeatures.remove();
+                        }
                         // check if the selected feature is contained in the features collection;
                         // if so then it is a conflict also and we need to clean it up and block it
                         String selectedFeature = selectedChain.getCandidates().get(0);
                         if (features.contains(selectedFeature)) {
-                            Chain conflictedFeatureChain = new Chain(Collections.<String> emptyList(), Collections.singletonList(featureSymbolicName), preferredVersion,
-                                                                     featureSymbolicName);
-                            addConflict(base, new ArrayList<Chain>(Arrays.asList(selectedChain, conflictedFeatureChain)));
+                            if (!allowedConflict) {
+                                Chain conflictedFeatureChain = new Chain(Collections.<String> emptyList(), Collections.singletonList(featureSymbolicName), preferredVersion,
+                                                                         featureSymbolicName);
+                                addConflict(base, new ArrayList<Chain>(Arrays.asList(selectedChain, conflictedFeatureChain)));
+                            }
                             conflicts.put(selectedFeature, base);
```

Comment:
We still add the selectedFeature to the conflicts even when `allowConflict` is true?

Reply:
The conflicts is local and is used to remove any conflicting features.

Reply:
Updated to not have this logic any longer and changed to use continue.